### PR TITLE
Ppereira all changes 1.24

### PR DESCRIPTION
**Pull request by princepereira**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/proxy/winkernel/proxier.go [hunk 1]
```diff
@@ -780,15 +805,26 @@ func CleanupLeftovers() (encounteredError bool) {
 	return encounteredError
 }
 
-func (svcInfo *serviceInfo) cleanupAllPolicies(endpoints []proxy.Endpoint) {
+func (svcInfo *serviceInfo) cleanupAllPolicies(endpoints []proxy.Endpoint, isEndpointChange bool) {
```
Comment (daschott) [inline pkg/proxy/winkernel/proxier.go:808]: *nit:* Should we rename "isEndpointChange" to "isEndpointTerminated"?

File: pkg/proxy/winkernel/proxier.go [hunk 2]
```diff
@@ -1124,9 +1213,16 @@ func (proxier *Proxier) syncProxyRules() {
 				continue
 			}
 
-			if !ep.IsReady() {
+			if !allEndpointsTerminating && !ep.IsReady() {
+				klog.V(3).InfoS("Skipping the endpoint for LB creation. Endpoint is either not ready or all not all endpoints are terminating", "EpIP", ep.ip, " EpPort", ep.port, "allEndpointsTerminating", allEndpointsTerminating, "IsEpReady", ep.IsReady())
```
Comment (daschott) [inline pkg/proxy/winkernel/proxier.go:1217]: This logging message is a bit confusing for me to read. Remove the "all not".

File: pkg/proxy/winkernel/hnsV1.go [hunk 3]
```diff
@@ -37,6 +37,7 @@ type HostNetworkService interface {
 	getEndpointByName(id string) (*endpointsInfo, error)
 	createEndpoint(ep *endpointsInfo, networkName string) (*endpointsInfo, error)
 	deleteEndpoint(hnsID string) error
+	findLoadBalancerID(endpoints []endpointsInfo, vip string, protocol, internalPort, externalPort uint16) (loadBalancerIdentifier, error)
```
[the remark this answers is not in the document]
Comment (daschott) [reply pkg/proxy/winkernel/hnsV1.go:40]: Can we also use the findLoadBalancerID in the getLoadBalancer method to remove the duplicated code?

[the remark this answers is not in the document]
Comment (princepereira) [reply pkg/proxy/winkernel/hnsV1.go]: Sure David. Will do. Thanks.

File: pkg/proxy/winkernel/proxier.go (hunk 1 above)
Comment (princepereira) [reply pkg/proxy/winkernel/proxier.go:820]: We wanted to inform here that the function is called as part of endpointmap change and not servicemap change

**Comment by dims** [comment]
This PR has the label work-in-progress, please revisit to see if you still need this, please close if not

**Comment by daschott** [comment]
lots of great fix-ups and changes here, thank you

**Pull request closed without merging**